os/drivers/audio/syu645b.c

```diff
@@ -0,0 +1,1055 @@
+ *
+ ****************************************************************************/
+static void syu645b_set_i2s_datawidth(FAR struct syu645b_dev_s *priv)
+{
+        /* if no audio device object return */
+        if (!priv) {
+                auddbg("Error, Device's private data Not available\n");
+                return;
+        }
+
+        if (priv->inout) {
+                I2S_RXDATAWIDTH(priv->i2s, priv->bpsamp);
+        } else {
+                I2S_TXDATAWIDTH(priv->i2s, priv->bpsamp);
+        }
+}
+
+/****************************************************************************
+ * Name: syu645b_set_i2s_samplerate
+ *
+ * Description:
+ *
+ ****************************************************************************/
+static void syu645b_set_i2s_samplerate(FAR struct syu645b_dev_s *priv)
+{
+        /* if no audio device object return */
+        if (!priv) {
+                auddbg("Error, Device's private data Not available\n");
+                return;
+        }
+
+        if (priv->inout) {
+                I2S_RXSAMPLERATE(priv->i2s, priv->samprate);
+        } else {
+                I2S_TXSAMPLERATE(priv->i2s, priv->samprate);
+        }
+}
+
+/****************************************************************************
+ * Name: syu645b_getcaps
+ *
+ * Description:
+ *   Get the audio device capabilities
+ *
+ ****************************************************************************/
+static int syu645b_getcaps(FAR struct audio_lowerhalf_s *dev, int type, FAR struct audio_caps_s *caps)
+{
+#if !defined(CONFIG_AUDIO_EXCLUDE_VOLUME) || !defined(CONFIG_AUDIO_EXCLUDE_TONE)
+	FAR struct syu645b_dev_s *priv = (FAR struct syu645b_dev_s *)dev;
+#endif
+	/* Validate the structure */
+
+	DEBUGASSERT(caps && caps->ac_len >= sizeof(struct audio_caps_s));
+	audvdbg("type=%d ac_type=%d\n", type, caps->ac_type);
+
+	/* Fill in the caller's structure based on requested info */
+
+	caps->ac_controls.w = 0;
+
+	switch (caps->ac_type) {
+	/* Caller is querying for the types of units we support */
+
+	case AUDIO_TYPE_QUERY:
+
+		/* Provide our overall capabilities.  The interfacing software
+		 * must then call us back for specific info for each capability.
+		 */
+
+		caps->ac_channels = SYU645B_CHANNELS;	/* Stereo output */
+
+		switch (caps->ac_subtype) {
+		case AUDIO_TYPE_QUERY:
+			/* We don't decode any formats!  Only something above us in
+			 * the audio stream can perform decoding on our behalf.
+			 */
+
+			/* The types of audio units we implement */
+			caps->ac_controls.b[0] = AUDIO_TYPE_OUTPUT | AUDIO_TYPE_FEATURE;
+
+			break;
+
+		default:
+			caps->ac_controls.b[0] = AUDIO_SUBFMT_END;
+			break;
+		}
+
+		break;
+
+	/* Provide capabilities of our INPUT & OUTPUT unit */
+	case AUDIO_TYPE_OUTPUT:
+		caps->ac_channels = SYU645B_CHANNELS;
+		switch (caps->ac_subtype) {
+		case AUDIO_TYPE_QUERY:
+			/* Report the Sample rates we support */
+			caps->ac_controls.b[0] = AUDIO_SAMP_RATE_TYPE_48K;
```

Comment:
it support more sample rate.
please check.

Reply:
@ritesh55555

Reply:
currently it is using 48K

Reply:
Let's expand later.